### PR TITLE
Add generic OIDC authentication

### LINKED ISSUE #5791: Add generic OIDC authentication for self-hosted deployments
## Problem

Self-hosted Multica deployments cannot use an organization-managed OpenID Connect provider such as Authentik or GitLab. Operators are limited to email verification codes or the dedicated Google OAuth integration, which prevents centralized access policy and group-based admission.

## Proposed behavior

- Add one discovery-capable generic OIDC provider alongside the existing Google flow.
- Validate authorization responses with state, nonce, PKCE, issuer, audience, signature, subject, and verified email checks.
- Link accounts by the stable `(issuer, subject)` identity so provider-side email changes do not create duplicate users.
- Optionally restrict access with a configurable group claim and allowlist.
- Support web, desktop, CLI-compatible callback state, and iOS browser handoff.
- Expose only the provider display name through public runtime config.
- Document environment variables and wire them through Docker Compose and Helm.

## Acceptance criteria

- Authentik and GitLab can be configured through standard OIDC discovery.
- Existing email-code and Google sign-in continue to work unchanged.
- OIDC is hidden when configuration is incomplete.
- Malformed or replayed flows fail closed.
- Deployment documentation and automated tests cover configuration, callback routing, group claims, and mobile handoff.

### PR DESCRIPTION
**Pull request by drybalka-s**
## What does this PR do?

Adds a separate, discovery-based OpenID Connect authentication path for self-hosted Multica deployments using providers such as Authentik and GitLab. Existing email-code and Google flows remain unchanged.

The provider exchange stays on the server. The flow validates state, nonce, PKCE, issuer, audience, signature, subject, and verified email, then links returning users by stable `(issuer, subject)` identity instead of mutable email.

## Related Issue

Closes #5791

## Type of Change

- [ ] Bug fix
- [x] New feature
- [ ] Refactor
- [x] Documentation update
- [x] Tests
- [ ] CI/CD or infrastructure change

## Changes Made

- Added generic OIDC discovery, authorization, callback, and signed flow-state handling.
- Added stable OIDC identity persistence and concurrent-index migrations.
- Added verified-email and optional group-allowlist enforcement.
- Wired provider discovery into web and desktop login.
- Added iOS system-browser handoff through `multica://auth/callback`.
- Added Docker Compose, Helm, environment, and operator documentation.
- Added coverage for configuration, state, nonce, PKCE, scalar and array group claims, callback routing, and mobile handoff.

## How to Test

1. Configure `OIDC_ISSUER_URL`, `OIDC_CLIENT_ID`, `OIDC_CLIENT_SECRET`, and `OIDC_REDIRECT_URI`.
2. Sign in through the configured provider on web, desktop, or mobile.
3. Set `OIDC_ALLOWED_GROUPS` and verify allowed and rejected memberships.
4. Change the provider-side email and verify the existing account is reused.

Automated validation completed:

- `go test ./...`
- web: 112 tests, typecheck, ESLint
- mobile: 35 tests, typecheck, lint
- `helm lint deploy/helm/multica`
- `docker compose -f docker-compose.selfhost.yml config --quiet`
- `git diff --check`

## Checklist

- [x] I have included a thinking path that traces from project context to this change
- [x] I have run tests locally and they pass
- [x] I have added or updated tests where applicable
- [ ] If this change affects the UI, I have included before/after screenshots
- [x] I have updated relevant documentation to reflect my changes
- [x] If this change touches Chinese product copy, I checked it against `apps/docs/content/docs/developers/conventions.zh.mdx`
- [x] I have considered and documented any risks above
- [x] I will address all reviewer comments before requesting merge

Risks and remaining manual checks:

- Live Authentik/GitLab end-to-end validation requires deployment credentials and was not run locally.
- iOS simulator visual validation was unavailable because Xcode `simctl` is not installed.
- Existing deployments must apply migrations 285-287 before OIDC logins.

## AI Disclosure

**AI tool used:** OpenAI Codex

**Approach:** Implemented and reviewed the integration iteratively against the repository auth, package-boundary, migration, mobile, deployment, and API compatibility rules. Added focused tests for discovered edge cases and ran the relevant Go, web, mobile, Helm, and Compose checks.

### CODE REVIEW

**Comment by AllenHu2024** [comment]
Fully support

**Comment by AllenHu2024** [comment]
Fully support!

**Comment by medoix** [comment]
+1 this would be a great option to remove the need for Resend and email based management.
i was looking to move Multica to our internal OIDC Dex server when i found this issue.

**Comment by sevenjay** [comment]
I tested the OIDC integration with Keycloak. After applying the following fix, it works as expected.

This fixes a PostgreSQL NUL-byte issue in the current OIDC branch in `server/internal/handler/oidc.go`.

The branch currently includes `\x00` in a PostgreSQL text parameter, which causes the following error:
`invalid byte sequence for encoding "UTF8": 0x00`

```diff
diff --git a/server/internal/handler/oidc.go b/server/internal/handler/oidc.go
index 0cf46f2d..8ad8bda0 100644
--- a/server/internal/handler/oidc.go
+++ b/server/internal/handler/oidc.go
@@ -334,7 +334,7 @@ func (h *Handler) resolveOIDCUser(ctx context.Context, issuer, subject, email st
        }
        defer func() { _ = tx.Rollback(ctx) }()
 
-       identityKey := issuer + "\x00" + subject
+       identityKey := "oidc-identity:" + issuer + "\x1f" + subject
        if _, err := tx.Exec(ctx, "SELECT pg_advisory_xact_lock(hashtextextended($1, 0))", identityKey); err != nil {
                return db.User{}, false, err
        }
@@ -352,7 +352,7 @@ func (h *Handler) resolveOIDCUser(ctx context.Context, issuer, subject, email st
        if !errors.Is(err, pgx.ErrNoRows) {
                return db.User{}, false, err
        }
-       if _, err := tx.Exec(ctx, "SELECT pg_advisory_xact_lock(hashtextextended($1, 0))", "oidc-email\x00"+email); err != nil {
+       if _, err := tx.Exec(ctx, "SELECT pg_advisory_xact_lock(hashtextextended($1, 0))", "oidc-email:"+email); err != nil {
                return db.User{}, false, err
        }
        user, isNew, err := h.findOrCreateUserWithQueries(ctx, queries, email)
```

**Comment by drybalka-s** [comment]
Thanks @sevenjay for testing this with Keycloak and for the precise fix. You were right: PostgreSQL text values cannot contain NUL bytes. I have applied the suggested advisory-lock key change in db8cda13a and pushed it to this PR.

**Comment by xiaoyaner0201** [comment]
I prepared a current-`main` refresh of this PR while preserving @drybalka-s's original 11 commits and authorship:

https://github.com/multica-ai/multica/compare/main...xiaoyaner0201:feat/refresh-generic-oidc-auth

The refresh is based on `main` at `11861145abc59a0d39c8c8f24ad837d4584e664f`. In addition to resolving the current-tree conflicts, it includes the following compatibility and security fixes:

- preserves the latest temporarily-disabled-user checks in the shared federated login path;
- keeps `user_oidc_identity` classified as global data in the workspace deletion manifest;
- renumbers the OIDC migrations to the next unique prefixes (`444`–`446`);
- registers interrupted concurrent-index cleanup for both OIDC indexes;
- makes the table migration retry-safe and fails closed on an incompatible pre-existing relation;
- keeps email and `email_verified` provenance bound to the same normalized identity claim;
- enforces exact, case-sensitive OIDC group allowlist matching;
- publishes the complete authenticated client-store state after OIDC login;
- limits OIDC to web and CLI browser flows for now. Native desktop/mobile OIDC handoff is intentionally excluded because a reusable session JWT must not be returned through an unclaimed custom URL scheme.

Verification on the refreshed exact tree:

- `go test -race ./internal/auth ./internal/handler ./internal/migrations ./cmd/migrate`
- `go vet ./internal/auth ./internal/handler ./internal/migrations ./cmd/migrate`
- `go tool govulncheck ./internal/auth ./internal/handler`
- retry/fail-closed migration integration test against PostgreSQL 17
- `go build ./cmd/server ./cmd/multica ./cmd/migrate`
- `pnpm typecheck`
- `pnpm lint`
- `pnpm build`
- focused Core, Views, and Web authentication tests
- `docker compose -f docker-compose.selfhost.yml config`
- `sqlc generate` with a clean generated-code diff

The full Go suite still has timing-sensitive failures under `server/pkg/agent` on this macOS host; the same package fails on a clean checkout of the pinned `main`, while all OIDC, migration, and handler packages pass.

Two possible handoff paths:

1. @drybalka-s can merge or cherry-pick the refresh into this PR; or
2. if the author or maintainers prefer, I can open an explicitly attributed Draft successor from the refreshed branch.

I will not open a parallel Ready PR or close this one without direction.
